lib/luwra/usertypes.hpp

```diff
@@ -119,8 +119,9 @@ namespace internal {
  */
 template <typename U>
 struct Value<U&> {
-	using T = internal::StripUserType<U>;
-
+	using X = internal::StripUserType<U>;
+	using T = typename std::conditional<std::is_function<U>::value && !std::is_pointer<U>::value, U*, U>::type;
```

Comment:
Ah ok :laughing: 
I am really tired, and all I could think is: "what is this black magic?!"